### DISCUSSION TITLE
support for compiling openwrt?

### DISCUSSION THREAD

**Discussion by mueslimak3r** [geerlingguy/raspberry-pi-pcie-devices#261] [Ideas] [answered]
long-time viewer of your youtube channel here. Looking forward to seeing that GPU working.

I am attempting to get openwrt working on my rapsberry pi 4B with a rtl8812AU wifi card. Unfortunately the drivers aren't officially stable or even working. But support can be enabled by compiling openwrt with some patches (ill link below).

Could the cross-compile environment be used for openwrt and/or other non-raspberry pi OS stuff?

additional context:
I'm trying to set up a layer 2 bridge between 2 buildings, using wifi. mainbuilding with dhcp and wifi->raspberrt pi wifi client/bridge running openwrt -> ethernet connection to an AP in the 2nd building

resources:
https://forum.openwrt.org/t/rtl8812au-and-or-rtl8814au-drivers/67
https://openwrt.org/docs/guide-user/network/wifi/atheroswds

**Comment by geerlingguy** [comment, marked as answer]
Moved this to a Discussion; at this time I don't think I want to try to support more than official Pi OS in the cross-compile environment in this project (though it's generic enough you just need to know the right make commands and repos to check out if you want to compile something else...).

But also have a look through the bottom of the DFRobot issue: https://github.com/geerlingguy/raspberry-pi-pcie-devices/issues/114

And I know someone had posted all the instructions for building a custom OpenWRT image at some point, but I'm having trouble finding that.

**Reply by mueslimak3r** [reply]
Thank! I'll give it a shot and see if your dev environment works for building openwrt
